### PR TITLE
Refactor command timer

### PR DESCRIPTION
**Pull request by MrPicklePinosaur**
added `try_get` and `try_get_mut` methods to state to reduces chances for runtime panics

### CODE REVIEW

File: crates/shrs_core/src/state.rs [hunk 1]
```diff
@@ -78,31 +80,91 @@ impl<T: 'static> DerefMut for StateMut<'_, T> {
...
 #[derive(Default)]
 pub struct States {
     states: HashMap<TypeId, RefCell<Box<dyn Any>>>,
 }
+
 impl States {
+
+    // Insert a new piece of state of given type into global state store, overriding previously
+    // existing values
+    // TODO should we allow overriding contents of state?
```
Comment (MrPicklePinosaur) [inline crates/shrs_core/src/state.rs:108]: @nithinmuthukumar thoughts?

File: crates/shrs_core/src/state.rs [hunk 2]
```diff
@@ -78,31 +80,91 @@ impl<T: 'static> DerefMut for StateMut<'_, T> {
...
-    pub fn remove<S>() {}
 
-    pub fn get_mut<S: 'static>(&self) -> RefMut<S> {
-        let s = self
-            .states
-            .get(&TypeId::of::<S>())
-            .expect("Value Missing")
-            .borrow_mut();
-        RefMut::map(s, |b| b.downcast_mut::<S>().unwrap())
+    // TODO this is potentially dangerous to allow arbitrary code to remove state
```
Comment (MrPicklePinosaur) [inline crates/shrs_core/src/state.rs:114]: @nithinmuthukumar thoughts on this too, should we even allow removal of state?

File: crates/shrs_core/src/state.rs (hunk 1 above)
Comment (nithinmuthukumar) [reply crates/shrs_core/src/state.rs:108]: Currently inserting a state of the same type overrides the current state.

File: crates/shrs_core/src/state.rs (hunk 2 above)
Comment (nithinmuthukumar) [reply crates/shrs_core/src/state.rs:114]: Think its fine to allow removal of state but it has to come with optional di params, bevy allows Option<Res<T>> which is something we could also allow in the future

Comment (nithinmuthukumar) [reply crates/shrs_core/src/state.rs:114]: We should also write in docs what types shouldnt be removed or add warnings if those types are removed

Comment (MrPicklePinosaur) [reply crates/shrs_core/src/state.rs:114]: optional state sounds cool, maybe lets open an issue to potentially implement that in the future? although i dont really see any uses cases of optional state currently

**Pull request merged**